Presentation compiler crashes on dependencies sources when Scala version is not 2.12.8

Potentially related to #474 

### Steps to reproduce:
- write `List` in a file
- go to definition from `List`
- go to the definition from `type List[+A] = scala.collection.immutable.List[A]`
- find references or go to definition from `def isEmpty: Boolean`

### Expected:
- the performed action succeeds 

### Actual:
- PC crashes
- If the attempted action is "go to definition", it prints also

```
code navigation does not work for the file '/Users/gabro/buildo/PROJECT/api/.metals/readonly/scala/collection/immutable/List.scala' because it doesn't belong to a build target.
```

Here it is:

<details><summary>Crash log</summary>
<p>

```
text document: file:///Users/gabro/buildo/PROJECT/api/.metals/readonly/scala/collection/immutable/List.scala
java.lang.AssertionError: assertion failed: 
  (method mapConserve,loop$default$1)
     while compiling: file:///Users/gabro/buildo/PROJECT/api/.metals/readonly/scala/collection/immutable/List.scala
        during phase: globalPhase=<no phase>, enteringPhase=typer
     library version: version 2.12.8
    compiler version: version 2.12.8
  reconstructed args: -Ywarn-value-discard -feature -Ypresentation-any-thread -opt-warnings:at-inline-failed -Ypartial-unification -Ymacro-expand:discard -Ywarn-numeric-widen -Ywarn-unused:imports -Ywarn-unused:patvars -Ywarn-unused:privates -Ywarn-unused:locals -Ywarn-unused:explicits -Ywarn-unused:implicits -unchecked -Xfuture -Ywarn-unused-import -language:implicitConversions -language:higherKinds -encoding utf8 -P:semanticdb:failures:warning -P:semanticdb:sourceroot:/Users/gabro/buildo/PROJECT/api -deprecation -classpath /Users/gabro/.sbt/boot/scala-2.12.4/lib/scala-library.jar:/Users/gabro/.ivy2/cache/io.buildo/wiro-http-server_2.12/jars/wiro-http-server_2.12-0.6.13.jar:/Users/gabro/.ivy2/cache/io.buildo/wiro-core_2.12/jars/wiro-core_2.12-0.6.13.jar:/Users/gabro/.sbt/boot/scala-2.12.4/lib/scala-reflect.jar:/Users/gabro/.ivy2/cache/com.lihaoyi/autowire_2.12/jars/autowire_2.12-0.2.6.jar:/Users/gabro/.ivy2/cache/com.typesafe.akka/akka-http_2.12/jars/akka-http_2.12-10.1.3.jar:/Users/gabro/.ivy2/cache/com.typesafe.akka/akka-http-core_2.12/jars/akka-http-core_2.12-10.1.3.jar:/Users/gabro/.ivy2/cache/com.typesafe.akka/akka-parsing_2.12/jars/akka-parsing_2.12-10.1.3.jar:/Users/gabro/.ivy2/cache/com.typesafe.akka/akka-actor_2.12/jars/akka-actor_2.12-2.5.14.jar:/Users/gabro/.ivy2/cache/com.typesafe/config/bundles/config-1.3.3.jar:/Users/gabro/.ivy2/cache/org.scala-lang.modules/scala-java8-compat_2.12/bundles/scala-java8-compat_2.12-0.8.0.jar:/Users/gabro/.ivy2/cache/com.typesafe.akka/akka-stream_2.12/jars/akka-stream_2.12-2.5.14.jar:/Users/gabro/.ivy2/cache/com.typesafe.akka/akka-protobuf_2.12/jars/akka-protobuf_2.12-2.5.14.jar:/Users/gabro/.ivy2/cache/org.reactivestreams/reactive-streams/jars/reactive-streams-1.0.2.jar:/Users/gabro/.ivy2/cache/com.typesafe/ssl-config-core_2.12/bundles/ssl-config-core_2.12-0.2.3.jar:/Users/gabro/.ivy2/cache/org.scala-lang.modules/scala-parser-combinators_2.12/bundles/scala-parser-combinators_2.12-1.1.0.jar:/Users/gabro/.ivy2/cache/de.heikoseeberger/akka-http-circe_2.12/jars/akka-http-circe_2.12-1.21.0.jar:/Users/gabro/.ivy2/cache/io.circe/circe-core_2.12/jars/circe-core_2.12-0.9.1.jar:/Users/gabro/.ivy2/cache/io.circe/circe-numbers_2.12/jars/circe-numbers_2.12-0.9.1.jar:/Users/gabro/.ivy2/cache/org.typelevel/cats-core_2.12/jars/cats-core_2.12-1.1.0.jar:/Users/gabro/.ivy2/cache/org.typelevel/cats-macros_2.12/jars/cats-macros_2.12-1.1.0.jar:/Users/gabro/.ivy2/cache/org.typelevel/machinist_2.12/jars/machinist_2.12-0.6.2.jar:/Users/gabro/.ivy2/cache/org.typelevel/cats-kernel_2.12/jars/cats-kernel_2.12-1.1.0.jar:/Users/gabro/.ivy2/cache/io.circe/circe-jawn_2.12/jars/circe-jawn_2.12-0.9.3.jar:/Users/gabro/.ivy2/cache/org.spire-math/jawn-parser_2.12/jars/jawn-parser_2.12-0.11.1.jar:/Users/gabro/.ivy2/cache/com.github.pureconfig/pureconfig_2.12/bundles/pureconfig_2.12-0.9.0.jar:/Users/gabro/.ivy2/cache/com.github.pureconfig/pureconfig-macros_2.12/jars/pureconfig-macros_2.12-0.9.0.jar:/Users/gabro/.sbt/boot/scala-2.12.4/lib/scala-compiler.jar:/Users/gabro/.ivy2/cache/org.scala-lang.modules/scala-xml_2.12/bundles/scala-xml_2.12-1.0.6.jar:/Users/gabro/.ivy2/cache/org.typelevel/macro-compat_2.12/jars/macro-compat_2.12-1.1.1.jar:/Users/gabro/.ivy2/cache/com.chuusai/shapeless_2.12/bundles/shapeless_2.12-2.3.3.jar:/Users/gabro/.ivy2/cache/io.circe/circe-parser_2.12/jars/circe-parser_2.12-0.9.0.jar:/Users/gabro/.ivy2/cache/io.circe/circe-generic-extras_2.12/jars/circe-generic-extras_2.12-0.9.1.jar:/Users/gabro/.ivy2/cache/io.circe/circe-generic_2.12/jars/circe-generic_2.12-0.9.1.jar:/Users/gabro/.ivy2/cache/io.buildo/java-time-circe-codecs_2.12/jars/java-time-circe-codecs_2.12-0.2.0.jar:/Users/gabro/.ivy2/cache/io.circe/circe-java8_2.12/jars/circe-java8_2.12-0.9.0.jar:/Users/gabro/.ivy2/cache/org.postgresql/postgresql/bundles/postgresql-42.2.1.jar:/Users/gabro/.ivy2/cache/io.getquill/quill-jdbc_2.12/jars/quill-jdbc_2.12-2.5.4.jar:/Users/gabro/.ivy2/cache/io.getquill/quill-sql_2.12/jars/quill-sql_2.12-2.5.4.jar:/Users/gabro/.ivy2/cache/io.getquill/quill-core_2.12/jars/quill-core_2.12-2.5.4.jar:/Users/gabro/.ivy2/cache/org.scalamacros/resetallattrs_2.12/jars/resetallattrs_2.12-1.0.0.jar:/Users/gabro/.ivy2/cache/com.zaxxer/HikariCP/bundles/HikariCP-2.7.4.jar:/Users/gabro/.ivy2/cache/io.getquill/quill-async-postgres_2.12/jars/quill-async-postgres_2.12-2.5.4.jar:/Users/gabro/.ivy2/cache/io.getquill/quill-async_2.12/jars/quill-async_2.12-2.5.4.jar:/Users/gabro/.ivy2/cache/com.github.mauricio/db-async-common_2.12/jars/db-async-common_2.12-0.2.21.jar:/Users/gabro/.ivy2/cache/joda-time/joda-time/jars/joda-time-2.9.7.jar:/Users/gabro/.ivy2/cache/org.joda/joda-convert/jars/joda-convert-1.8.1.jar:/Users/gabro/.ivy2/cache/io.netty/netty-all/jars/netty-all-4.1.6.Final.jar:/Users/gabro/.ivy2/cache/org.javassist/javassist/bundles/javassist-3.21.0-GA.jar:/Users/gabro/.ivy2/cache/com.github.mauricio/postgresql-async_2.12/jars/postgresql-async_2.12-0.2.21.jar:/Users/gabro/.ivy2/cache/org.flywaydb/flyway-core/jars/flyway-core-5.0.7.jar:/Users/gabro/.ivy2/cache/io.buildo/toctoc-core_2.12/jars/toctoc-core_2.12-0.1.9.jar:/Users/gabro/.ivy2/cache/org.mindrot/jbcrypt/jars/jbcrypt-0.4.jar:/Users/gabro/.ivy2/cache/io.buildo/enumero_2.12/jars/enumero_2.12-1.3.0.jar:/Users/gabro/.ivy2/cache/io.buildo/toctoc-quill_2.12/jars/toctoc-quill_2.12-0.1.9.jar:/Users/gabro/.ivy2/cache/org.bouncycastle/bcpg-jdk15on/jars/bcpg-jdk15on-1.60.jar:/Users/gabro/.ivy2/cache/org.bouncycastle/bcprov-jdk15on/jars/bcprov-jdk15on-1.60.jar:/Users/gabro/.ivy2/cache/com.sun.mail/javax.mail/jars/javax.mail-1.6.1.jar:/Users/gabro/.ivy2/cache/javax.activation/activation/jars/activation-1.1.jar:/Users/gabro/.ivy2/cache/org.bouncycastle/bcpkix-jdk15on/jars/bcpkix-jdk15on-1.60.jar:/Users/gabro/.ivy2/cache/io.buildo/enumero-circe-support_2.12/jars/enumero-circe-support_2.12-1.3.0.jar:/Users/gabro/.ivy2/cache/io.buildo/mailo_2.12/jars/mailo_2.12-0.4.4.jar:/Users/gabro/.ivy2/cache/com.github.seratch/awscala_2.12/jars/awscala_2.12-0.5.9.jar:/Users/gabro/.ivy2/cache/com.amazonaws/aws-java-sdk-iam/jars/aws-java-sdk-iam-1.10.77.jar:/Users/gabro/.ivy2/cache/commons-logging/commons-logging/jars/commons-logging-1.1.3.jar:/Users/gabro/.ivy2/cache/com.amazonaws/aws-java-sdk-sts/jars/aws-java-sdk-sts-1.10.77.jar:/Users/gabro/.ivy2/cache/com.amazonaws/aws-java-sdk-ec2/jars/aws-java-sdk-ec2-1.10.77.jar:/Users/gabro/.ivy2/cache/com.amazonaws/aws-java-sdk-s3/jars/aws-java-sdk-s3-1.11.339.jar:/Users/gabro/.ivy2/cache/com.amazonaws/aws-java-sdk-kms/jars/aws-java-sdk-kms-1.11.339.jar:/Users/gabro/.ivy2/cache/com.amazonaws/aws-java-sdk-core/jars/aws-java-sdk-core-1.11.339.jar:/Users/gabro/.ivy2/cache/org.apache.httpcomponents/httpclient/jars/httpclient-4.5.5.jar:/Users/gabro/.ivy2/cache/org.apache.httpcomponents/httpcore/jars/httpcore-4.4.9.jar:/Users/gabro/.ivy2/cache/commons-codec/commons-codec/jars/commons-codec-1.10.jar:/Users/gabro/.ivy2/cache/software.amazon.ion/ion-java/bundles/ion-java-1.0.2.jar:/Users/gabro/.ivy2/cache/com.fasterxml.jackson.core/jackson-databind/bundles/jackson-databind-2.6.7.1.jar:/Users/gabro/.ivy2/cache/com.fasterxml.jackson.core/jackson-annotations/bundles/jackson-annotations-2.6.0.jar:/Users/gabro/.ivy2/cache/com.fasterxml.jackson.core/jackson-core/bundles/jackson-core-2.6.7.jar:/Users/gabro/.ivy2/cache/com.fasterxml.jackson.dataformat/jackson-dataformat-cbor/bundles/jackson-dataformat-cbor-2.6.7.jar:/Users/gabro/.ivy2/cache/com.amazonaws/jmespath-java/jars/jmespath-java-1.11.339.jar:/Users/gabro/.ivy2/cache/com.amazonaws/aws-java-sdk-sqs/jars/aws-java-sdk-sqs-1.10.77.jar:/Users/gabro/.ivy2/cache/com.amazonaws/aws-java-sdk-emr/jars/aws-java-sdk-emr-1.10.77.jar:/Users/gabro/.ivy2/cache/com.amazonaws/aws-java-sdk-redshift/jars/aws-java-sdk-redshift-1.10.77.jar:/Users/gabro/.ivy2/cache/com.amazonaws/aws-java-sdk-dynamodb/jars/aws-java-sdk-dynamodb-1.10.77.jar:/Users/gabro/.ivy2/cache/com.amazonaws/aws-java-sdk-simpledb/jars/aws-java-sdk-simpledb-1.10.77.jar:/Users/gabro/.ivy2/cache/org.typelevel/alleycats-core_2.12/jars/alleycats-core_2.12-1.0.1.jar:/Users/gabro/.ivy2/cache/org.typelevel/export-hook_2.12/jars/export-hook_2.12-1.2.0.jar:/Users/gabro/.ivy2/cache/com.github.cb372/scalacache-guava_2.12/jars/scalacache-guava_2.12-0.9.3.jar:/Users/gabro/.ivy2/cache/com.github.cb372/scalacache-core_2.12/jars/scalacache-core_2.12-0.9.3.jar:/Users/gabro/.ivy2/cache/com.google.guava/guava/bundles/guava-19.0.jar:/Users/gabro/.ivy2/cache/com.google.code.findbugs/jsr305/jars/jsr305-1.3.9.jar:/Users/gabro/.ivy2/cache/com.typesafe.scala-logging/scala-logging_2.12/bundles/scala-logging_2.12-3.8.0.jar:/Users/gabro/.ivy2/cache/javax.mail/javax.mail-api/jars/javax.mail-api-1.6.1.jar:/Users/gabro/.ivy2/cache/org.apache.logging.log4j/log4j-api/jars/log4j-api-2.11.0.jar:/Users/gabro/.ivy2/cache/org.apache.logging.log4j/log4j-api-scala_2.12/jars/log4j-api-scala_2.12-11.0.jar:/Users/gabro/.ivy2/cache/org.apache.logging.log4j/log4j-slf4j-impl/jars/log4j-slf4j-impl-2.11.0.jar:/Users/gabro/.ivy2/cache/org.slf4j/slf4j-api/jars/slf4j-api-1.8.0-alpha2.jar:/Users/gabro/.ivy2/cache/org.apache.logging.log4j/log4j-core/jars/log4j-core-2.11.0.jar:/Users/gabro/.ivy2/cache/com.lightbend.akka/akka-stream-alpakka-s3_2.12/jars/akka-stream-alpakka-s3_2.12-0.19.jar:/Users/gabro/.ivy2/cache/com.typesafe.akka/akka-http-xml_2.12/jars/akka-http-xml_2.12-10.0.13.jar:/Users/gabro/.ivy2/cache/com.lihaoyi/sourcecode_2.12/bundles/sourcecode_2.12-0.1.4.jar -Yrangepos -Ywarn-dead-code -Xlint:adapted-args -Xlint:nullary-unit -Xlint:inaccessible -Xlint:nullary-override -Xlint:infer-any -Xlint:missing-interpolator -Xlint:doc-detached -Xlint:private-shadow -Xlint:type-parameter-shadow -Xlint:poly-implicit-overload -Xlint:option-implicit -Xlint:delayedinit-select -Xlint:by-name-right-associative -Xlint:package-object-classes -Xlint:unsound-match -Xlint:stars-align -Xlint:constant -Xlint:unused -Xplugin:/Users/gabro/.ivy2/cache/org.scalamacros/paradise_2.12.4/jars/paradise_2.12.4-2.1.0.jar -Xplugin:/Users/gabro/.ivy2/cache/org.scalameta/semanticdb-scalac_2.12.4/jars/semanticdb-scalac_2.12.4-4.1.0.jar -Xplugin-require:semanticdb

  last tree to typer: Ident(B)
       tree position: line 166 of file:///Users/gabro/buildo/PROJECT/api/.metals/readonly/scala/collection/immutable/List.scala
            tree tpe: B
              symbol: type B
   symbol definition: B >: A <: AnyRef (a TypeSkolem)
      symbol package: scala.collection.immutable
       symbol owners: type B -> method mapConserve -> class List
           call site: <none> in <none>

== Source file context for tree position ==

   163     // Note to developers: there exists a duplication between this function and `reflect.internal.util.Collections#map2Conserve`.
   164     // If any successful optimization attempts or other changes are made, please rehash them there too.
   165     @tailrec
   166     def loop(mappedHead: List[B] = Nil, mappedLast: ::[B], unchanged: List[A], pending: List[A]): List[B] =
   167     if (pending.isEmpty) {
   168       if (mappedHead eq null) unchanged
   169       else {
	at scala.reflect.internal.SymbolTable.throwAssertionError(SymbolTable.scala:183)
	at scala.tools.nsc.typechecker.Namers$Namer.$anonfun$addDefaultGetters$10(Namers.scala:1588)
	at scala.tools.nsc.typechecker.Namers$Namer$DefaultMethodInOwningScope.addGetter(Namers.scala:1664)
	at scala.tools.nsc.typechecker.Namers$Namer.$anonfun$addDefaultGetters$8(Namers.scala:1557)
	at scala.tools.nsc.typechecker.Namers$Namer.$anonfun$addDefaultGetters$6(Namers.scala:1528)
	at scala.tools.nsc.typechecker.Namers$Namer.addDefaultGetters(Namers.scala:1524)
	at scala.tools.nsc.typechecker.Namers$Namer.methodSig(Namers.scala:1411)
	at scala.tools.nsc.typechecker.Namers$Namer.memberSig(Namers.scala:1876)
	at scala.tools.nsc.typechecker.Namers$Namer.typeSig(Namers.scala:1842)
	at scala.tools.nsc.typechecker.Namers$Namer$MonoTypeCompleter.completeImpl(Namers.scala:853)
	at scala.tools.nsc.typechecker.Namers$LockingTypeCompleter.complete(Namers.scala:2039)
	at scala.tools.nsc.typechecker.Namers$LockingTypeCompleter.complete$(Namers.scala:2037)
	at scala.tools.nsc.typechecker.Namers$TypeCompleterBase.complete(Namers.scala:2032)
	at scala.reflect.internal.Symbols$Symbol.info(Symbols.scala:1542)
	at scala.reflect.internal.Symbols$Symbol.initialize(Symbols.scala:1690)
	at scala.tools.nsc.typechecker.Typers$Typer.typed1(Typers.scala:5349)
	at scala.tools.nsc.typechecker.Typers$Typer.typed(Typers.scala:5726)
	at scala.tools.nsc.typechecker.Typers$Typer.typedStat$1(Typers.scala:5790)
	at scala.tools.nsc.typechecker.Typers$Typer.$anonfun$typedStats$10(Typers.scala:3373)
	at scala.tools.nsc.typechecker.Typers$Typer.typedStats(Typers.scala:3373)
	at scala.tools.nsc.typechecker.Typers$Typer.typedBlock(Typers.scala:2503)
	at scala.tools.nsc.typechecker.Typers$Typer.$anonfun$typed1$98(Typers.scala:5655)
	at scala.tools.nsc.typechecker.Typers$Typer.typedOutsidePatternMode$1(Typers.scala:500)
	at scala.tools.nsc.typechecker.Typers$Typer.typed1(Typers.scala:5690)
	at scala.tools.nsc.typechecker.Typers$Typer.typed(Typers.scala:5726)
	at scala.tools.nsc.typechecker.Typers$Typer.typedDefDef(Typers.scala:5942)
	at scala.tools.nsc.typechecker.Typers$Typer.typed1(Typers.scala:5645)
	at scala.tools.nsc.typechecker.Typers$Typer.typed(Typers.scala:5726)
	at scala.tools.nsc.typechecker.Typers$Typer.typedStat$1(Typers.scala:5790)
	at scala.tools.nsc.typechecker.Typers$Typer.$anonfun$typedStats$10(Typers.scala:3373)
	at scala.tools.nsc.typechecker.Typers$Typer.typedStats(Typers.scala:3373)
	at scala.tools.nsc.typechecker.Typers$Typer.typedTemplate(Typers.scala:2030)
	at scala.tools.nsc.typechecker.Typers$Typer.typedClassDef(Typers.scala:1843)
	at scala.tools.nsc.typechecker.Typers$Typer.typed1(Typers.scala:5646)
	at scala.tools.nsc.typechecker.Typers$Typer.typed(Typers.scala:5726)
	at scala.tools.nsc.typechecker.Typers$Typer.typedStat$1(Typers.scala:5790)
	at scala.tools.nsc.typechecker.Typers$Typer.$anonfun$typedStats$10(Typers.scala:3373)
	at scala.tools.nsc.typechecker.Typers$Typer.typedStats(Typers.scala:3373)
	at scala.tools.nsc.typechecker.Typers$Typer.typedPackageDef$1(Typers.scala:5356)
	at scala.tools.nsc.typechecker.Typers$Typer.typed1(Typers.scala:5649)
	at scala.tools.nsc.typechecker.Typers$Typer.typed(Typers.scala:5726)
	at scala.tools.nsc.typechecker.Typers$Typer.typedStat$1(Typers.scala:5790)
	at scala.tools.nsc.typechecker.Typers$Typer.$anonfun$typedStats$10(Typers.scala:3373)
	at scala.tools.nsc.typechecker.Typers$Typer.typedStats(Typers.scala:3373)
	at scala.tools.nsc.typechecker.Typers$Typer.typedPackageDef$1(Typers.scala:5356)
	at scala.tools.nsc.typechecker.Typers$Typer.typed1(Typers.scala:5649)
	at scala.tools.nsc.typechecker.Typers$Typer.typed(Typers.scala:5726)
	at scala.tools.nsc.typechecker.Typers$Typer.typedStat$1(Typers.scala:5790)
	at scala.tools.nsc.typechecker.Typers$Typer.$anonfun$typedStats$10(Typers.scala:3373)
	at scala.tools.nsc.typechecker.Typers$Typer.typedStats(Typers.scala:3373)
	at scala.tools.nsc.typechecker.Typers$Typer.typedPackageDef$1(Typers.scala:5356)
	at scala.tools.nsc.typechecker.Typers$Typer.typed1(Typers.scala:5649)
	at scala.tools.nsc.typechecker.Typers$Typer.typed(Typers.scala:5726)
	at scala.tools.nsc.typechecker.Analyzer$typerFactory$TyperPhase.apply(Analyzer.scala:114)
	at scala.tools.nsc.Global$GlobalPhase.applyPhase(Global.scala:448)
	at scala.tools.nsc.interactive.Global$TyperRun.$anonfun$applyPhase$1(Global.scala:1340)
	at scala.tools.nsc.interactive.Global$TyperRun.applyPhase(Global.scala:1340)
	at scala.tools.nsc.interactive.Global$TyperRun.typeCheck(Global.scala:1333)
	at scala.tools.nsc.interactive.Global.typeCheck(Global.scala:665)
	at scala.tools.nsc.interactive.Global.$anonfun$backgroundCompile$7(Global.scala:580)
	at scala.tools.nsc.interactive.Global.$anonfun$backgroundCompile$6(Global.scala:576)
	at scala.tools.nsc.interactive.Global.$anonfun$backgroundCompile$6$adapted(Global.scala:576)
	at scala.collection.TraversableLike$WithFilter.$anonfun$foreach$1(TraversableLike.scala:792)
	at scala.collection.immutable.List.foreach(List.scala:392)
	at scala.collection.TraversableLike$WithFilter.foreach(TraversableLike.scala:791)
	at scala.tools.nsc.interactive.Global.backgroundCompile(Global.scala:576)
	at scala.tools.nsc.interactive.PresentationCompilerThread.run(PresentationCompilerThread.scala:32)

code navigation does not work for the file '/Users/gabro/buildo/PROJECT/api/.metals/readonly/scala/collection/immutable/List.scala' because it doesn't belong to a build target.
text document: file:///Users/gabro/buildo/PROJECT/api/.metals/readonly/scala/collection/immutable/List.scala
java.lang.AssertionError: assertion failed: 
  (method mapConserve,loop$default$1)
     while compiling: file:///Users/gabro/buildo/PROJECT/api/.metals/readonly/scala/collection/immutable/List.scala
        during phase: globalPhase=<no phase>, enteringPhase=typer
     library version: version 2.12.8
    compiler version: version 2.12.8
  reconstructed args: -Ywarn-value-discard -feature -Ypresentation-any-thread -opt-warnings:at-inline-failed -Ypartial-unification -Ymacro-expand:discard -Ywarn-numeric-widen -Ywarn-unused:imports -Ywarn-unused:patvars -Ywarn-unused:privates -Ywarn-unused:locals -Ywarn-unused:explicits -Ywarn-unused:implicits -unchecked -Xfuture -Ywarn-unused-import -language:implicitConversions -language:higherKinds -encoding utf8 -P:semanticdb:failures:warning -P:semanticdb:sourceroot:/Users/gabro/buildo/PROJECT/api -deprecation -classpath /Users/gabro/.sbt/boot/scala-2.12.4/lib/scala-library.jar:/Users/gabro/.ivy2/cache/io.buildo/wiro-http-server_2.12/jars/wiro-http-server_2.12-0.6.13.jar:/Users/gabro/.ivy2/cache/io.buildo/wiro-core_2.12/jars/wiro-core_2.12-0.6.13.jar:/Users/gabro/.sbt/boot/scala-2.12.4/lib/scala-reflect.jar:/Users/gabro/.ivy2/cache/com.lihaoyi/autowire_2.12/jars/autowire_2.12-0.2.6.jar:/Users/gabro/.ivy2/cache/com.typesafe.akka/akka-http_2.12/jars/akka-http_2.12-10.1.3.jar:/Users/gabro/.ivy2/cache/com.typesafe.akka/akka-http-core_2.12/jars/akka-http-core_2.12-10.1.3.jar:/Users/gabro/.ivy2/cache/com.typesafe.akka/akka-parsing_2.12/jars/akka-parsing_2.12-10.1.3.jar:/Users/gabro/.ivy2/cache/com.typesafe.akka/akka-actor_2.12/jars/akka-actor_2.12-2.5.14.jar:/Users/gabro/.ivy2/cache/com.typesafe/config/bundles/config-1.3.3.jar:/Users/gabro/.ivy2/cache/org.scala-lang.modules/scala-java8-compat_2.12/bundles/scala-java8-compat_2.12-0.8.0.jar:/Users/gabro/.ivy2/cache/com.typesafe.akka/akka-stream_2.12/jars/akka-stream_2.12-2.5.14.jar:/Users/gabro/.ivy2/cache/com.typesafe.akka/akka-protobuf_2.12/jars/akka-protobuf_2.12-2.5.14.jar:/Users/gabro/.ivy2/cache/org.reactivestreams/reactive-streams/jars/reactive-streams-1.0.2.jar:/Users/gabro/.ivy2/cache/com.typesafe/ssl-config-core_2.12/bundles/ssl-config-core_2.12-0.2.3.jar:/Users/gabro/.ivy2/cache/org.scala-lang.modules/scala-parser-combinators_2.12/bundles/scala-parser-combinators_2.12-1.1.0.jar:/Users/gabro/.ivy2/cache/de.heikoseeberger/akka-http-circe_2.12/jars/akka-http-circe_2.12-1.21.0.jar:/Users/gabro/.ivy2/cache/io.circe/circe-core_2.12/jars/circe-core_2.12-0.9.1.jar:/Users/gabro/.ivy2/cache/io.circe/circe-numbers_2.12/jars/circe-numbers_2.12-0.9.1.jar:/Users/gabro/.ivy2/cache/org.typelevel/cats-core_2.12/jars/cats-core_2.12-1.1.0.jar:/Users/gabro/.ivy2/cache/org.typelevel/cats-macros_2.12/jars/cats-macros_2.12-1.1.0.jar:/Users/gabro/.ivy2/cache/org.typelevel/machinist_2.12/jars/machinist_2.12-0.6.2.jar:/Users/gabro/.ivy2/cache/org.typelevel/cats-kernel_2.12/jars/cats-kernel_2.12-1.1.0.jar:/Users/gabro/.ivy2/cache/io.circe/circe-jawn_2.12/jars/circe-jawn_2.12-0.9.3.jar:/Users/gabro/.ivy2/cache/org.spire-math/jawn-parser_2.12/jars/jawn-parser_2.12-0.11.1.jar:/Users/gabro/.ivy2/cache/com.github.pureconfig/pureconfig_2.12/bundles/pureconfig_2.12-0.9.0.jar:/Users/gabro/.ivy2/cache/com.github.pureconfig/pureconfig-macros_2.12/jars/pureconfig-macros_2.12-0.9.0.jar:/Users/gabro/.sbt/boot/scala-2.12.4/lib/scala-compiler.jar:/Users/gabro/.ivy2/cache/org.scala-lang.modules/scala-xml_2.12/bundles/scala-xml_2.12-1.0.6.jar:/Users/gabro/.ivy2/cache/org.typelevel/macro-compat_2.12/jars/macro-compat_2.12-1.1.1.jar:/Users/gabro/.ivy2/cache/com.chuusai/shapeless_2.12/bundles/shapeless_2.12-2.3.3.jar:/Users/gabro/.ivy2/cache/io.circe/circe-parser_2.12/jars/circe-parser_2.12-0.9.0.jar:/Users/gabro/.ivy2/cache/io.circe/circe-generic-extras_2.12/jars/circe-generic-extras_2.12-0.9.1.jar:/Users/gabro/.ivy2/cache/io.circe/circe-generic_2.12/jars/circe-generic_2.12-0.9.1.jar:/Users/gabro/.ivy2/cache/io.buildo/java-time-circe-codecs_2.12/jars/java-time-circe-codecs_2.12-0.2.0.jar:/Users/gabro/.ivy2/cache/io.circe/circe-java8_2.12/jars/circe-java8_2.12-0.9.0.jar:/Users/gabro/.ivy2/cache/org.postgresql/postgresql/bundles/postgresql-42.2.1.jar:/Users/gabro/.ivy2/cache/io.getquill/quill-jdbc_2.12/jars/quill-jdbc_2.12-2.5.4.jar:/Users/gabro/.ivy2/cache/io.getquill/quill-sql_2.12/jars/quill-sql_2.12-2.5.4.jar:/Users/gabro/.ivy2/cache/io.getquill/quill-core_2.12/jars/quill-core_2.12-2.5.4.jar:/Users/gabro/.ivy2/cache/org.scalamacros/resetallattrs_2.12/jars/resetallattrs_2.12-1.0.0.jar:/Users/gabro/.ivy2/cache/com.zaxxer/HikariCP/bundles/HikariCP-2.7.4.jar:/Users/gabro/.ivy2/cache/io.getquill/quill-async-postgres_2.12/jars/quill-async-postgres_2.12-2.5.4.jar:/Users/gabro/.ivy2/cache/io.getquill/quill-async_2.12/jars/quill-async_2.12-2.5.4.jar:/Users/gabro/.ivy2/cache/com.github.mauricio/db-async-common_2.12/jars/db-async-common_2.12-0.2.21.jar:/Users/gabro/.ivy2/cache/joda-time/joda-time/jars/joda-time-2.9.7.jar:/Users/gabro/.ivy2/cache/org.joda/joda-convert/jars/joda-convert-1.8.1.jar:/Users/gabro/.ivy2/cache/io.netty/netty-all/jars/netty-all-4.1.6.Final.jar:/Users/gabro/.ivy2/cache/org.javassist/javassist/bundles/javassist-3.21.0-GA.jar:/Users/gabro/.ivy2/cache/com.github.mauricio/postgresql-async_2.12/jars/postgresql-async_2.12-0.2.21.jar:/Users/gabro/.ivy2/cache/org.flywaydb/flyway-core/jars/flyway-core-5.0.7.jar:/Users/gabro/.ivy2/cache/io.buildo/toctoc-core_2.12/jars/toctoc-core_2.12-0.1.9.jar:/Users/gabro/.ivy2/cache/org.mindrot/jbcrypt/jars/jbcrypt-0.4.jar:/Users/gabro/.ivy2/cache/io.buildo/enumero_2.12/jars/enumero_2.12-1.3.0.jar:/Users/gabro/.ivy2/cache/io.buildo/toctoc-quill_2.12/jars/toctoc-quill_2.12-0.1.9.jar:/Users/gabro/.ivy2/cache/org.bouncycastle/bcpg-jdk15on/jars/bcpg-jdk15on-1.60.jar:/Users/gabro/.ivy2/cache/org.bouncycastle/bcprov-jdk15on/jars/bcprov-jdk15on-1.60.jar:/Users/gabro/.ivy2/cache/com.sun.mail/javax.mail/jars/javax.mail-1.6.1.jar:/Users/gabro/.ivy2/cache/javax.activation/activation/jars/activation-1.1.jar:/Users/gabro/.ivy2/cache/org.bouncycastle/bcpkix-jdk15on/jars/bcpkix-jdk15on-1.60.jar:/Users/gabro/.ivy2/cache/io.buildo/enumero-circe-support_2.12/jars/enumero-circe-support_2.12-1.3.0.jar:/Users/gabro/.ivy2/cache/io.buildo/mailo_2.12/jars/mailo_2.12-0.4.4.jar:/Users/gabro/.ivy2/cache/com.github.seratch/awscala_2.12/jars/awscala_2.12-0.5.9.jar:/Users/gabro/.ivy2/cache/com.amazonaws/aws-java-sdk-iam/jars/aws-java-sdk-iam-1.10.77.jar:/Users/gabro/.ivy2/cache/commons-logging/commons-logging/jars/commons-logging-1.1.3.jar:/Users/gabro/.ivy2/cache/com.amazonaws/aws-java-sdk-sts/jars/aws-java-sdk-sts-1.10.77.jar:/Users/gabro/.ivy2/cache/com.amazonaws/aws-java-sdk-ec2/jars/aws-java-sdk-ec2-1.10.77.jar:/Users/gabro/.ivy2/cache/com.amazonaws/aws-java-sdk-s3/jars/aws-java-sdk-s3-1.11.339.jar:/Users/gabro/.ivy2/cache/com.amazonaws/aws-java-sdk-kms/jars/aws-java-sdk-kms-1.11.339.jar:/Users/gabro/.ivy2/cache/com.amazonaws/aws-java-sdk-core/jars/aws-java-sdk-core-1.11.339.jar:/Users/gabro/.ivy2/cache/org.apache.httpcomponents/httpclient/jars/httpclient-4.5.5.jar:/Users/gabro/.ivy2/cache/org.apache.httpcomponents/httpcore/jars/httpcore-4.4.9.jar:/Users/gabro/.ivy2/cache/commons-codec/commons-codec/jars/commons-codec-1.10.jar:/Users/gabro/.ivy2/cache/software.amazon.ion/ion-java/bundles/ion-java-1.0.2.jar:/Users/gabro/.ivy2/cache/com.fasterxml.jackson.core/jackson-databind/bundles/jackson-databind-2.6.7.1.jar:/Users/gabro/.ivy2/cache/com.fasterxml.jackson.core/jackson-annotations/bundles/jackson-annotations-2.6.0.jar:/Users/gabro/.ivy2/cache/com.fasterxml.jackson.core/jackson-core/bundles/jackson-core-2.6.7.jar:/Users/gabro/.ivy2/cache/com.fasterxml.jackson.dataformat/jackson-dataformat-cbor/bundles/jackson-dataformat-cbor-2.6.7.jar:/Users/gabro/.ivy2/cache/com.amazonaws/jmespath-java/jars/jmespath-java-1.11.339.jar:/Users/gabro/.ivy2/cache/com.amazonaws/aws-java-sdk-sqs/jars/aws-java-sdk-sqs-1.10.77.jar:/Users/gabro/.ivy2/cache/com.amazonaws/aws-java-sdk-emr/jars/aws-java-sdk-emr-1.10.77.jar:/Users/gabro/.ivy2/cache/com.amazonaws/aws-java-sdk-redshift/jars/aws-java-sdk-redshift-1.10.77.jar:/Users/gabro/.ivy2/cache/com.amazonaws/aws-java-sdk-dynamodb/jars/aws-java-sdk-dynamodb-1.10.77.jar:/Users/gabro/.ivy2/cache/com.amazonaws/aws-java-sdk-simpledb/jars/aws-java-sdk-simpledb-1.10.77.jar:/Users/gabro/.ivy2/cache/org.typelevel/alleycats-core_2.12/jars/alleycats-core_2.12-1.0.1.jar:/Users/gabro/.ivy2/cache/org.typelevel/export-hook_2.12/jars/export-hook_2.12-1.2.0.jar:/Users/gabro/.ivy2/cache/com.github.cb372/scalacache-guava_2.12/jars/scalacache-guava_2.12-0.9.3.jar:/Users/gabro/.ivy2/cache/com.github.cb372/scalacache-core_2.12/jars/scalacache-core_2.12-0.9.3.jar:/Users/gabro/.ivy2/cache/com.google.guava/guava/bundles/guava-19.0.jar:/Users/gabro/.ivy2/cache/com.google.code.findbugs/jsr305/jars/jsr305-1.3.9.jar:/Users/gabro/.ivy2/cache/com.typesafe.scala-logging/scala-logging_2.12/bundles/scala-logging_2.12-3.8.0.jar:/Users/gabro/.ivy2/cache/javax.mail/javax.mail-api/jars/javax.mail-api-1.6.1.jar:/Users/gabro/.ivy2/cache/org.apache.logging.log4j/log4j-api/jars/log4j-api-2.11.0.jar:/Users/gabro/.ivy2/cache/org.apache.logging.log4j/log4j-api-scala_2.12/jars/log4j-api-scala_2.12-11.0.jar:/Users/gabro/.ivy2/cache/org.apache.logging.log4j/log4j-slf4j-impl/jars/log4j-slf4j-impl-2.11.0.jar:/Users/gabro/.ivy2/cache/org.slf4j/slf4j-api/jars/slf4j-api-1.8.0-alpha2.jar:/Users/gabro/.ivy2/cache/org.apache.logging.log4j/log4j-core/jars/log4j-core-2.11.0.jar:/Users/gabro/.ivy2/cache/com.lightbend.akka/akka-stream-alpakka-s3_2.12/jars/akka-stream-alpakka-s3_2.12-0.19.jar:/Users/gabro/.ivy2/cache/com.typesafe.akka/akka-http-xml_2.12/jars/akka-http-xml_2.12-10.0.13.jar:/Users/gabro/.ivy2/cache/com.lihaoyi/sourcecode_2.12/bundles/sourcecode_2.12-0.1.4.jar -Yrangepos -Ywarn-dead-code -Xlint:adapted-args -Xlint:nullary-unit -Xlint:inaccessible -Xlint:nullary-override -Xlint:infer-any -Xlint:missing-interpolator -Xlint:doc-detached -Xlint:private-shadow -Xlint:type-parameter-shadow -Xlint:poly-implicit-overload -Xlint:option-implicit -Xlint:delayedinit-select -Xlint:by-name-right-associative -Xlint:package-object-classes -Xlint:unsound-match -Xlint:stars-align -Xlint:constant -Xlint:unused -Xplugin:/Users/gabro/.ivy2/cache/org.scalamacros/paradise_2.12.4/jars/paradise_2.12.4-2.1.0.jar -Xplugin:/Users/gabro/.ivy2/cache/org.scalameta/semanticdb-scalac_2.12.4/jars/semanticdb-scalac_2.12.4-4.1.0.jar -Xplugin-require:semanticdb

  last tree to typer: Ident(B)
       tree position: line 166 of file:///Users/gabro/buildo/PROJECT/api/.metals/readonly/scala/collection/immutable/List.scala
            tree tpe: B
              symbol: type B
   symbol definition: B >: A <: AnyRef (a TypeSkolem)
      symbol package: scala.collection.immutable
       symbol owners: type B -> method mapConserve -> class List
           call site: <none> in <none>

== Source file context for tree position ==

   163     // Note to developers: there exists a duplication between this function and `reflect.internal.util.Collections#map2Conserve`.
   164     // If any successful optimization attempts or other changes are made, please rehash them there too.
   165     @tailrec
   166     def loop(mappedHead: List[B] = Nil, mappedLast: ::[B], unchanged: List[A], pending: List[A]): List[B] =
   167     if (pending.isEmpty) {
   168       if (mappedHead eq null) unchanged
   169       else {
	at scala.reflect.internal.SymbolTable.throwAssertionError(SymbolTable.scala:183)
	at scala.tools.nsc.typechecker.Namers$Namer.$anonfun$addDefaultGetters$10(Namers.scala:1588)
	at scala.tools.nsc.typechecker.Namers$Namer$DefaultMethodInOwningScope.addGetter(Namers.scala:1664)
	at scala.tools.nsc.typechecker.Namers$Namer.$anonfun$addDefaultGetters$8(Namers.scala:1557)
	at scala.tools.nsc.typechecker.Namers$Namer.$anonfun$addDefaultGetters$6(Namers.scala:1528)
	at scala.tools.nsc.typechecker.Namers$Namer.addDefaultGetters(Namers.scala:1524)
	at scala.tools.nsc.typechecker.Namers$Namer.methodSig(Namers.scala:1411)
	at scala.tools.nsc.typechecker.Namers$Namer.memberSig(Namers.scala:1876)
	at scala.tools.nsc.typechecker.Namers$Namer.typeSig(Namers.scala:1842)
	at scala.tools.nsc.typechecker.Namers$Namer$MonoTypeCompleter.completeImpl(Namers.scala:853)
	at scala.tools.nsc.typechecker.Namers$LockingTypeCompleter.complete(Namers.scala:2039)
	at scala.tools.nsc.typechecker.Namers$LockingTypeCompleter.complete$(Namers.scala:2037)
	at scala.tools.nsc.typechecker.Namers$TypeCompleterBase.complete(Namers.scala:2032)
	at scala.reflect.internal.Symbols$Symbol.info(Symbols.scala:1542)
	at scala.reflect.internal.Symbols$Symbol.initialize(Symbols.scala:1690)
	at scala.tools.nsc.typechecker.Typers$Typer.typed1(Typers.scala:5349)
	at scala.tools.nsc.typechecker.Typers$Typer.typed(Typers.scala:5726)
	at scala.tools.nsc.typechecker.Typers$Typer.typedStat$1(Typers.scala:5790)
	at scala.tools.nsc.typechecker.Typers$Typer.$anonfun$typedStats$10(Typers.scala:3373)
	at scala.tools.nsc.typechecker.Typers$Typer.typedStats(Typers.scala:3373)
	at scala.tools.nsc.typechecker.Typers$Typer.typedBlock(Typers.scala:2503)
	at scala.tools.nsc.typechecker.Typers$Typer.$anonfun$typed1$98(Typers.scala:5655)
	at scala.tools.nsc.typechecker.Typers$Typer.typedOutsidePatternMode$1(Typers.scala:500)
	at scala.tools.nsc.typechecker.Typers$Typer.typed1(Typers.scala:5690)
	at scala.tools.nsc.typechecker.Typers$Typer.typed(Typers.scala:5726)
	at scala.tools.nsc.typechecker.Typers$Typer.typedDefDef(Typers.scala:5942)
	at scala.tools.nsc.typechecker.Typers$Typer.typed1(Typers.scala:5645)
	at scala.tools.nsc.typechecker.Typers$Typer.typed(Typers.scala:5726)
	at scala.tools.nsc.typechecker.Typers$Typer.typedStat$1(Typers.scala:5790)
	at scala.tools.nsc.typechecker.Typers$Typer.$anonfun$typedStats$10(Typers.scala:3373)
	at scala.tools.nsc.typechecker.Typers$Typer.typedStats(Typers.scala:3373)
	at scala.tools.nsc.typechecker.Typers$Typer.typedTemplate(Typers.scala:2030)
	at scala.tools.nsc.typechecker.Typers$Typer.typedClassDef(Typers.scala:1843)
	at scala.tools.nsc.typechecker.Typers$Typer.typed1(Typers.scala:5646)
	at scala.tools.nsc.typechecker.Typers$Typer.typed(Typers.scala:5726)
	at scala.tools.nsc.typechecker.Typers$Typer.typedStat$1(Typers.scala:5790)
	at scala.tools.nsc.typechecker.Typers$Typer.$anonfun$typedStats$10(Typers.scala:3373)
	at scala.tools.nsc.typechecker.Typers$Typer.typedStats(Typers.scala:3373)
	at scala.tools.nsc.typechecker.Typers$Typer.typedPackageDef$1(Typers.scala:5356)
	at scala.tools.nsc.typechecker.Typers$Typer.typed1(Typers.scala:5649)
	at scala.tools.nsc.typechecker.Typers$Typer.typed(Typers.scala:5726)
	at scala.tools.nsc.typechecker.Typers$Typer.typedStat$1(Typers.scala:5790)
	at scala.tools.nsc.typechecker.Typers$Typer.$anonfun$typedStats$10(Typers.scala:3373)
	at scala.tools.nsc.typechecker.Typers$Typer.typedStats(Typers.scala:3373)
	at scala.tools.nsc.typechecker.Typers$Typer.typedPackageDef$1(Typers.scala:5356)
	at scala.tools.nsc.typechecker.Typers$Typer.typed1(Typers.scala:5649)
	at scala.tools.nsc.typechecker.Typers$Typer.typed(Typers.scala:5726)
	at scala.tools.nsc.typechecker.Typers$Typer.typedStat$1(Typers.scala:5790)
	at scala.tools.nsc.typechecker.Typers$Typer.$anonfun$typedStats$10(Typers.scala:3373)
	at scala.tools.nsc.typechecker.Typers$Typer.typedStats(Typers.scala:3373)
	at scala.tools.nsc.typechecker.Typers$Typer.typedPackageDef$1(Typers.scala:5356)
	at scala.tools.nsc.typechecker.Typers$Typer.typed1(Typers.scala:5649)
	at scala.tools.nsc.typechecker.Typers$Typer.typed(Typers.scala:5726)
	at scala.tools.nsc.typechecker.Analyzer$typerFactory$TyperPhase.apply(Analyzer.scala:114)
	at scala.tools.nsc.Global$GlobalPhase.applyPhase(Global.scala:448)
	at scala.tools.nsc.interactive.Global$TyperRun.$anonfun$applyPhase$1(Global.scala:1340)
	at scala.tools.nsc.interactive.Global$TyperRun.applyPhase(Global.scala:1340)
	at scala.tools.nsc.interactive.Global$TyperRun.typeCheck(Global.scala:1333)
	at scala.tools.nsc.interactive.Global.typeCheck(Global.scala:665)
	at scala.tools.nsc.interactive.Global.$anonfun$backgroundCompile$7(Global.scala:580)
	at scala.tools.nsc.interactive.Global.$anonfun$backgroundCompile$6(Global.scala:576)
	at scala.tools.nsc.interactive.Global.$anonfun$backgroundCompile$6$adapted(Global.scala:576)
	at scala.collection.TraversableLike$WithFilter.$anonfun$foreach$1(TraversableLike.scala:792)
	at scala.collection.immutable.List.foreach(List.scala:392)
	at scala.collection.TraversableLike$WithFilter.foreach(TraversableLike.scala:791)
	at scala.tools.nsc.interactive.Global.backgroundCompile(Global.scala:576)
	at scala.tools.nsc.interactive.PresentationCompilerThread.run(PresentationCompilerThread.scala:32)

code navigation does not work for the file '/Users/gabro/buildo/PROJECT/api/.metals/readonly/scala/collection/immutable/List.scala' because it doesn't belong to a build target.
```
</p>
</details>

This happens after exactly the same steps above. After the first jump I could navigate fine, but after jumping again it crashed.

More to the discussion:

- set `serverVersion` to the latest master (`0.3.3+141-6fa9ec53-SNAPSHOT`)
- clone this toy project https://github.com/gabro/config
- import the build
- wait for compilation to end
- open `Authorization.scala`
- go to definition from `OptionT`
- go to definition from anything
- PC crashes

<details><summary>Stack trace</summary>
<p>

```
text document: file:///Users/gabro/buildo/config-scala/.metals/readonly/cats/data/OptionT.scala
java.lang.AssertionError: assertion failed: 
  (class Kleisli,copy$default$1)
     while compiling: file:///Users/gabro/buildo/config-scala/.metals/readonly/cats/data/Kleisli.scala
        during phase: globalPhase=<no phase>, enteringPhase=namer
     library version: version 2.12.8
    compiler version: version 2.12.8
  reconstructed args: -classpath /Users/gabro/.ivy2/cache/org.scala-lang/scala-library/jars/scala-library-2.12.7.jar:/Users/gabro/.ivy2/cache/org.http4s/http4s-dsl_2.12/jars/http4s-dsl_2.12-0.20.0-M2.jar:/Users/gabro/.ivy2/cache/org.http4s/http4s-core_2.12/jars/http4s-core_2.12-0.20.0-M2.jar:/Users/gabro/.ivy2/cache/org.typelevel/cats-core_2.12/jars/cats-core_2.12-1.4.0.jar:/Users/gabro/.ivy2/cache/org.typelevel/cats-macros_2.12/jars/cats-macros_2.12-1.4.0.jar:/Users/gabro/.ivy2/cache/org.typelevel/machinist_2.12/jars/machinist_2.12-0.6.5.jar:/Users/gabro/.ivy2/cache/org.scala-lang/scala-reflect/jars/scala-reflect-2.12.7.jar:/Users/gabro/.ivy2/cache/org.typelevel/cats-kernel_2.12/jars/cats-kernel_2.12-1.4.0.jar:/Users/gabro/.ivy2/cache/org.typelevel/cats-effect_2.12/jars/cats-effect_2.12-1.0.0.jar:/Users/gabro/.ivy2/cache/co.fs2/fs2-io_2.12/jars/fs2-io_2.12-1.0.0.jar:/Users/gabro/.ivy2/cache/co.fs2/fs2-core_2.12/jars/fs2-core_2.12-1.0.0.jar:/Users/gabro/.ivy2/cache/org.scodec/scodec-bits_2.12/bundles/scodec-bits_2.12-1.1.5.jar:/Users/gabro/.ivy2/cache/org.log4s/log4s_2.12/jars/log4s_2.12-1.6.1.jar:/Users/gabro/.ivy2/cache/org.slf4j/slf4j-api/jars/slf4j-api-1.7.25.jar:/Users/gabro/.ivy2/cache/org.http4s/parboiled_2.12/jars/parboiled_2.12-1.0.0.jar:/Users/gabro/.ivy2/cache/org.http4s/http4s-blaze-server_2.12/jars/http4s-blaze-server_2.12-0.20.0-M2.jar:/Users/gabro/.ivy2/cache/org.http4s/http4s-blaze-core_2.12/jars/http4s-blaze-core_2.12-0.20.0-M2.jar:/Users/gabro/.ivy2/cache/org.http4s/blaze-http_2.12/jars/blaze-http_2.12-0.14.0-M10.jar:/Users/gabro/.ivy2/cache/org.http4s/blaze-core_2.12/jars/blaze-core_2.12-0.14.0-M10.jar:/Users/gabro/.ivy2/cache/com.twitter/hpack/jars/hpack-1.0.2.jar:/Users/gabro/.ivy2/cache/org.eclipse.jetty.alpn/alpn-api/jars/alpn-api-1.1.3.v20160715.jar:/Users/gabro/.ivy2/cache/org.http4s/http4s-server_2.12/jars/http4s-server_2.12-0.20.0-M2.jar:/Users/gabro/.ivy2/cache/org.http4s/http4s-circe_2.12/jars/http4s-circe_2.12-0.20.0-M2.jar:/Users/gabro/.ivy2/cache/org.http4s/http4s-jawn_2.12/jars/http4s-jawn_2.12-0.20.0-M2.jar:/Users/gabro/.ivy2/cache/org.http4s/jawn-fs2_2.12/jars/jawn-fs2_2.12-0.13.0.jar:/Users/gabro/.ivy2/cache/org.spire-math/jawn-parser_2.12/jars/jawn-parser_2.12-0.13.0.jar:/Users/gabro/.ivy2/cache/io.circe/circe-jawn_2.12/jars/circe-jawn_2.12-0.10.1.jar:/Users/gabro/.ivy2/cache/io.circe/circe-core_2.12/jars/circe-core_2.12-0.10.1.jar:/Users/gabro/.ivy2/cache/io.circe/circe-numbers_2.12/jars/circe-numbers_2.12-0.10.1.jar:/Users/gabro/.ivy2/cache/io.circe/circe-generic_2.12/jars/circe-generic_2.12-0.10.0.jar:/Users/gabro/.ivy2/cache/com.chuusai/shapeless_2.12/bundles/shapeless_2.12-2.3.3.jar:/Users/gabro/.ivy2/cache/org.typelevel/macro-compat_2.12/jars/macro-compat_2.12-1.1.1.jar:/Users/gabro/.ivy2/cache/io.estatico/newtype_2.12/jars/newtype_2.12-0.4.2.jar:/Users/gabro/.ivy2/cache/io.chrisdavenport/fuuid_2.12/jars/fuuid_2.12-0.2.0-M2.jar:/Users/gabro/.ivy2/cache/io.chrisdavenport/fuuid-http4s_2.12/jars/fuuid-http4s_2.12-0.2.0-M2.jar:/Users/gabro/.ivy2/cache/io.chrisdavenport/fuuid-circe_2.12/jars/fuuid-circe_2.12-0.2.0-M2.jar:/Users/gabro/.ivy2/cache/io.chrisdavenport/log4cats-slf4j_2.12/jars/log4cats-slf4j_2.12-0.2.0.jar:/Users/gabro/.ivy2/cache/io.chrisdavenport/log4cats-slf4j-internal_2.12/jars/log4cats-slf4j-internal_2.12-0.2.0.jar:/Users/gabro/.ivy2/cache/io.chrisdavenport/log4cats-core_2.12/jars/log4cats-core_2.12-0.2.0.jar:/Users/gabro/.ivy2/cache/ch.qos.logback/logback-classic/jars/logback-classic-1.2.3.jar:/Users/gabro/.ivy2/cache/ch.qos.logback/logback-core/jars/logback-core-1.2.3.jar:/Users/gabro/.ivy2/cache/com.olegpy/meow-mtl_2.12/jars/meow-mtl_2.12-0.2.0.jar:/Users/gabro/.ivy2/cache/org.typelevel/cats-mtl-core_2.12/jars/cats-mtl-core_2.12-0.4.0.jar:/Users/gabro/.ivy2/cache/com.github.mpilquist/simulacrum_2.12/jars/simulacrum_2.12-0.12.0.jar:/Users/gabro/.ivy2/cache/org.scalacheck/scalacheck_2.12/jars/scalacheck_2.12-1.13.5.jar:/Users/gabro/.ivy2/cache/org.scala-sbt/test-interface/jars/test-interface-1.0.jar -Xplugin:/Users/gabro/.ivy2/cache/org.spire-math/kind-projector_2.12/jars/kind-projector_2.12-0.9.6.jar -Xplugin:/Users/gabro/.ivy2/cache/org.scalamacros/paradise_2.12.7/jars/paradise_2.12.7-2.1.1.jar -Xplugin:/Users/gabro/.ivy2/cache/org.scalameta/semanticdb-scalac_2.12.7/jars/semanticdb-scalac_2.12.7-4.1.0.jar -Yrangepos -P:semanticdb:synthetics:on -P:semanticdb:failures:warning -P:semanticdb:sourceroot:/Users/gabro/buildo/config-scala -Ypresentation-any-thread -Ypartial-unification -Ymacro-expand:discard -Xplugin-require:semanticdb

  last tree to typer: This(class Kleisli)
       tree position: line 10 of file:///Users/gabro/buildo/config-scala/.metals/readonly/cats/data/Kleisli.scala
            tree tpe: Kleisli.this.type
              symbol: (final case) class Kleisli in package data
   symbol definition: final case class Kleisli[F[_], A, B] extends Product with Serializable (a ClassSymbol)
      symbol package: cats.data
       symbol owners: class Kleisli
           call site: <none> in <none>

== Source file context for tree position ==

     7 /**
     8  * Represents a function `A => F[B]`.
     9  */
    10 final case class Kleisli[F[_], A, B](run: A => F[B]) { self =>
    11 
    12   def ap[C](f: Kleisli[F, A, B => C])(implicit F: Apply[F]): Kleisli[F, A, C] =
    13     Kleisli(a => F.ap(f.run(a))(run(a)))
	at scala.reflect.internal.SymbolTable.throwAssertionError(SymbolTable.scala:183)
	at scala.tools.nsc.typechecker.Namers$Namer.$anonfun$addDefaultGetters$10(Namers.scala:1588)
	at scala.tools.nsc.typechecker.Namers$Namer$DefaultMethodInOwningScope.addGetter(Namers.scala:1664)
	at scala.tools.nsc.typechecker.Namers$Namer.$anonfun$addDefaultGetters$8(Namers.scala:1557)
	at scala.tools.nsc.typechecker.Namers$Namer.$anonfun$addDefaultGetters$6(Namers.scala:1528)
	at scala.tools.nsc.typechecker.Namers$Namer.addDefaultGetters(Namers.scala:1524)
	at scala.tools.nsc.typechecker.Namers$Namer.methodSig(Namers.scala:1411)
	at scala.tools.nsc.typechecker.Namers$Namer.memberSig(Namers.scala:1876)
	at scala.tools.nsc.typechecker.Namers$Namer.typeSig(Namers.scala:1842)
	at scala.tools.nsc.typechecker.Namers$Namer$MonoTypeCompleter.completeImpl(Namers.scala:853)
	at scala.tools.nsc.typechecker.Namers$LockingTypeCompleter.complete(Namers.scala:2039)
	at scala.tools.nsc.typechecker.Namers$LockingTypeCompleter.complete$(Namers.scala:2037)
	at scala.tools.nsc.typechecker.Namers$TypeCompleterBase.complete(Namers.scala:2032)
	at scala.tools.nsc.typechecker.Namers$PolyTypeCompleter.completeImpl(Namers.scala:2072)
	at scala.tools.nsc.typechecker.Namers$LockingTypeCompleter.complete(Namers.scala:2039)
	at scala.tools.nsc.typechecker.Namers$LockingTypeCompleter.complete$(Namers.scala:2037)
	at scala.tools.nsc.typechecker.Namers$PolyTypeCompleter.complete(Namers.scala:2051)
	at scala.tools.nsc.typechecker.Namers$CompleterWrapper.complete(Namers.scala:2088)
	at scala.tools.nsc.typechecker.Namers$Namer$$anon$1.complete(Namers.scala:616)
	at scala.reflect.internal.Symbols$Symbol.info(Symbols.scala:1542)
	at scala.reflect.internal.Symbols$Symbol.initialize(Symbols.scala:1690)
	at scala.tools.nsc.typechecker.Typers$Typer.$anonfun$typedStats$6(Typers.scala:3325)
	at scala.tools.nsc.typechecker.Typers$Typer.$anonfun$typedStats$6$adapted(Typers.scala:3323)
	at scala.Option$WithFilter.foreach(Option.scala:274)
	at scala.tools.nsc.typechecker.Typers$Typer.$anonfun$typedStats$4(Typers.scala:3323)
	at scala.tools.nsc.typechecker.Typers$Typer.$anonfun$typedStats$4$adapted(Typers.scala:3321)
	at scala.reflect.internal.Scopes$Scope.foreach(Scopes.scala:415)
	at scala.tools.nsc.typechecker.Typers$Typer.addSynthetics$1(Typers.scala:3321)
	at scala.tools.nsc.typechecker.Typers$Typer.typedStats(Typers.scala:3383)
	at scala.tools.nsc.typechecker.Typers$Typer.typedTemplate(Typers.scala:2030)
	at scala.tools.nsc.typechecker.Typers$Typer.typedClassDef(Typers.scala:1843)
	at scala.tools.nsc.typechecker.Typers$Typer.typed1(Typers.scala:5646)
	at scala.tools.nsc.typechecker.Typers$Typer.typed(Typers.scala:5726)
	at scala.tools.nsc.typechecker.Typers$Typer.typedStat$1(Typers.scala:5790)
	at scala.tools.nsc.typechecker.Typers$Typer.$anonfun$typedStats$10(Typers.scala:3373)
	at scala.tools.nsc.typechecker.Typers$Typer.typedStats(Typers.scala:3373)
	at scala.tools.nsc.typechecker.Typers$Typer.typedPackageDef$1(Typers.scala:5356)
	at scala.tools.nsc.typechecker.Typers$Typer.typed1(Typers.scala:5649)
	at scala.tools.nsc.typechecker.Typers$Typer.typed(Typers.scala:5726)
	at scala.tools.nsc.typechecker.Typers$Typer.typedStat$1(Typers.scala:5790)
	at scala.tools.nsc.typechecker.Typers$Typer.$anonfun$typedStats$10(Typers.scala:3373)
	at scala.tools.nsc.typechecker.Typers$Typer.typedStats(Typers.scala:3373)
	at scala.tools.nsc.typechecker.Typers$Typer.typedPackageDef$1(Typers.scala:5356)
	at scala.tools.nsc.typechecker.Typers$Typer.typed1(Typers.scala:5649)
	at scala.tools.nsc.typechecker.Typers$Typer.typed(Typers.scala:5726)
	at scala.tools.nsc.typechecker.Analyzer$typerFactory$TyperPhase.apply(Analyzer.scala:114)
	at scala.tools.nsc.Global$GlobalPhase.applyPhase(Global.scala:448)
	at scala.tools.nsc.interactive.Global$TyperRun.$anonfun$applyPhase$1(Global.scala:1340)
	at scala.tools.nsc.interactive.Global$TyperRun.applyPhase(Global.scala:1340)
	at scala.tools.nsc.interactive.Global$TyperRun.typeCheck(Global.scala:1333)
	at scala.tools.nsc.interactive.Global.typeCheck(Global.scala:665)
	at scala.tools.nsc.interactive.Global.$anonfun$backgroundCompile$7(Global.scala:580)
	at scala.tools.nsc.interactive.Global.$anonfun$backgroundCompile$6(Global.scala:576)
	at scala.tools.nsc.interactive.Global.$anonfun$backgroundCompile$6$adapted(Global.scala:576)
	at scala.collection.TraversableLike$WithFilter.$anonfun$foreach$1(TraversableLike.scala:792)
	at scala.collection.immutable.List.foreach(List.scala:392)
	at scala.collection.TraversableLike$WithFilter.foreach(TraversableLike.scala:791)
	at scala.tools.nsc.interactive.Global.backgroundCompile(Global.scala:576)
	at scala.tools.nsc.interactive.PresentationCompilerThread.run(PresentationCompilerThread.scala:32)

code navigation does not work for the file '/Users/gabro/buildo/config-scala/.metals/readonly/cats/data/OptionT.scala' because it doesn't belong to a build target.
text document: file:///Users/gabro/buildo/config-scala/.metals/readonly/cats/data/OptionT.scala
java.lang.AssertionError: assertion failed: 
  (class Kleisli,copy$default$1)
     while compiling: file:///Users/gabro/buildo/config-scala/.metals/readonly/cats/data/Kleisli.scala
        during phase: globalPhase=<no phase>, enteringPhase=namer
     library version: version 2.12.8
    compiler version: version 2.12.8
  reconstructed args: -classpath /Users/gabro/.ivy2/cache/org.scala-lang/scala-library/jars/scala-library-2.12.7.jar:/Users/gabro/.ivy2/cache/org.http4s/http4s-dsl_2.12/jars/http4s-dsl_2.12-0.20.0-M2.jar:/Users/gabro/.ivy2/cache/org.http4s/http4s-core_2.12/jars/http4s-core_2.12-0.20.0-M2.jar:/Users/gabro/.ivy2/cache/org.typelevel/cats-core_2.12/jars/cats-core_2.12-1.4.0.jar:/Users/gabro/.ivy2/cache/org.typelevel/cats-macros_2.12/jars/cats-macros_2.12-1.4.0.jar:/Users/gabro/.ivy2/cache/org.typelevel/machinist_2.12/jars/machinist_2.12-0.6.5.jar:/Users/gabro/.ivy2/cache/org.scala-lang/scala-reflect/jars/scala-reflect-2.12.7.jar:/Users/gabro/.ivy2/cache/org.typelevel/cats-kernel_2.12/jars/cats-kernel_2.12-1.4.0.jar:/Users/gabro/.ivy2/cache/org.typelevel/cats-effect_2.12/jars/cats-effect_2.12-1.0.0.jar:/Users/gabro/.ivy2/cache/co.fs2/fs2-io_2.12/jars/fs2-io_2.12-1.0.0.jar:/Users/gabro/.ivy2/cache/co.fs2/fs2-core_2.12/jars/fs2-core_2.12-1.0.0.jar:/Users/gabro/.ivy2/cache/org.scodec/scodec-bits_2.12/bundles/scodec-bits_2.12-1.1.5.jar:/Users/gabro/.ivy2/cache/org.log4s/log4s_2.12/jars/log4s_2.12-1.6.1.jar:/Users/gabro/.ivy2/cache/org.slf4j/slf4j-api/jars/slf4j-api-1.7.25.jar:/Users/gabro/.ivy2/cache/org.http4s/parboiled_2.12/jars/parboiled_2.12-1.0.0.jar:/Users/gabro/.ivy2/cache/org.http4s/http4s-blaze-server_2.12/jars/http4s-blaze-server_2.12-0.20.0-M2.jar:/Users/gabro/.ivy2/cache/org.http4s/http4s-blaze-core_2.12/jars/http4s-blaze-core_2.12-0.20.0-M2.jar:/Users/gabro/.ivy2/cache/org.http4s/blaze-http_2.12/jars/blaze-http_2.12-0.14.0-M10.jar:/Users/gabro/.ivy2/cache/org.http4s/blaze-core_2.12/jars/blaze-core_2.12-0.14.0-M10.jar:/Users/gabro/.ivy2/cache/com.twitter/hpack/jars/hpack-1.0.2.jar:/Users/gabro/.ivy2/cache/org.eclipse.jetty.alpn/alpn-api/jars/alpn-api-1.1.3.v20160715.jar:/Users/gabro/.ivy2/cache/org.http4s/http4s-server_2.12/jars/http4s-server_2.12-0.20.0-M2.jar:/Users/gabro/.ivy2/cache/org.http4s/http4s-circe_2.12/jars/http4s-circe_2.12-0.20.0-M2.jar:/Users/gabro/.ivy2/cache/org.http4s/http4s-jawn_2.12/jars/http4s-jawn_2.12-0.20.0-M2.jar:/Users/gabro/.ivy2/cache/org.http4s/jawn-fs2_2.12/jars/jawn-fs2_2.12-0.13.0.jar:/Users/gabro/.ivy2/cache/org.spire-math/jawn-parser_2.12/jars/jawn-parser_2.12-0.13.0.jar:/Users/gabro/.ivy2/cache/io.circe/circe-jawn_2.12/jars/circe-jawn_2.12-0.10.1.jar:/Users/gabro/.ivy2/cache/io.circe/circe-core_2.12/jars/circe-core_2.12-0.10.1.jar:/Users/gabro/.ivy2/cache/io.circe/circe-numbers_2.12/jars/circe-numbers_2.12-0.10.1.jar:/Users/gabro/.ivy2/cache/io.circe/circe-generic_2.12/jars/circe-generic_2.12-0.10.0.jar:/Users/gabro/.ivy2/cache/com.chuusai/shapeless_2.12/bundles/shapeless_2.12-2.3.3.jar:/Users/gabro/.ivy2/cache/org.typelevel/macro-compat_2.12/jars/macro-compat_2.12-1.1.1.jar:/Users/gabro/.ivy2/cache/io.estatico/newtype_2.12/jars/newtype_2.12-0.4.2.jar:/Users/gabro/.ivy2/cache/io.chrisdavenport/fuuid_2.12/jars/fuuid_2.12-0.2.0-M2.jar:/Users/gabro/.ivy2/cache/io.chrisdavenport/fuuid-http4s_2.12/jars/fuuid-http4s_2.12-0.2.0-M2.jar:/Users/gabro/.ivy2/cache/io.chrisdavenport/fuuid-circe_2.12/jars/fuuid-circe_2.12-0.2.0-M2.jar:/Users/gabro/.ivy2/cache/io.chrisdavenport/log4cats-slf4j_2.12/jars/log4cats-slf4j_2.12-0.2.0.jar:/Users/gabro/.ivy2/cache/io.chrisdavenport/log4cats-slf4j-internal_2.12/jars/log4cats-slf4j-internal_2.12-0.2.0.jar:/Users/gabro/.ivy2/cache/io.chrisdavenport/log4cats-core_2.12/jars/log4cats-core_2.12-0.2.0.jar:/Users/gabro/.ivy2/cache/ch.qos.logback/logback-classic/jars/logback-classic-1.2.3.jar:/Users/gabro/.ivy2/cache/ch.qos.logback/logback-core/jars/logback-core-1.2.3.jar:/Users/gabro/.ivy2/cache/com.olegpy/meow-mtl_2.12/jars/meow-mtl_2.12-0.2.0.jar:/Users/gabro/.ivy2/cache/org.typelevel/cats-mtl-core_2.12/jars/cats-mtl-core_2.12-0.4.0.jar:/Users/gabro/.ivy2/cache/com.github.mpilquist/simulacrum_2.12/jars/simulacrum_2.12-0.12.0.jar:/Users/gabro/.ivy2/cache/org.scalacheck/scalacheck_2.12/jars/scalacheck_2.12-1.13.5.jar:/Users/gabro/.ivy2/cache/org.scala-sbt/test-interface/jars/test-interface-1.0.jar -Xplugin:/Users/gabro/.ivy2/cache/org.spire-math/kind-projector_2.12/jars/kind-projector_2.12-0.9.6.jar -Xplugin:/Users/gabro/.ivy2/cache/org.scalamacros/paradise_2.12.7/jars/paradise_2.12.7-2.1.1.jar -Xplugin:/Users/gabro/.ivy2/cache/org.scalameta/semanticdb-scalac_2.12.7/jars/semanticdb-scalac_2.12.7-4.1.0.jar -Yrangepos -P:semanticdb:synthetics:on -P:semanticdb:failures:warning -P:semanticdb:sourceroot:/Users/gabro/buildo/config-scala -Ypresentation-any-thread -Ypartial-unification -Ymacro-expand:discard -Xplugin-require:semanticdb

  last tree to typer: This(class Kleisli)
       tree position: line 10 of file:///Users/gabro/buildo/config-scala/.metals/readonly/cats/data/Kleisli.scala
            tree tpe: Kleisli.this.type
              symbol: (final case) class Kleisli in package data
   symbol definition: final case class Kleisli[F[_], A, B] extends Product with Serializable (a ClassSymbol)
      symbol package: cats.data
       symbol owners: class Kleisli
           call site: <none> in <none>

== Source file context for tree position ==

     7 /**
     8  * Represents a function `A => F[B]`.
     9  */
    10 final case class Kleisli[F[_], A, B](run: A => F[B]) { self =>
    11 
    12   def ap[C](f: Kleisli[F, A, B => C])(implicit F: Apply[F]): Kleisli[F, A, C] =
    13     Kleisli(a => F.ap(f.run(a))(run(a)))
	at scala.reflect.internal.SymbolTable.throwAssertionError(SymbolTable.scala:183)
	at scala.tools.nsc.typechecker.Namers$Namer.$anonfun$addDefaultGetters$10(Namers.scala:1588)
	at scala.tools.nsc.typechecker.Namers$Namer$DefaultMethodInOwningScope.addGetter(Namers.scala:1664)
	at scala.tools.nsc.typechecker.Namers$Namer.$anonfun$addDefaultGetters$8(Namers.scala:1557)
	at scala.tools.nsc.typechecker.Namers$Namer.$anonfun$addDefaultGetters$6(Namers.scala:1528)
	at scala.tools.nsc.typechecker.Namers$Namer.addDefaultGetters(Namers.scala:1524)
	at scala.tools.nsc.typechecker.Namers$Namer.methodSig(Namers.scala:1411)
	at scala.tools.nsc.typechecker.Namers$Namer.memberSig(Namers.scala:1876)
	at scala.tools.nsc.typechecker.Namers$Namer.typeSig(Namers.scala:1842)
	at scala.tools.nsc.typechecker.Namers$Namer$MonoTypeCompleter.completeImpl(Namers.scala:853)
	at scala.tools.nsc.typechecker.Namers$LockingTypeCompleter.complete(Namers.scala:2039)
	at scala.tools.nsc.typechecker.Namers$LockingTypeCompleter.complete$(Namers.scala:2037)
	at scala.tools.nsc.typechecker.Namers$TypeCompleterBase.complete(Namers.scala:2032)
	at scala.tools.nsc.typechecker.Namers$PolyTypeCompleter.completeImpl(Namers.scala:2072)
	at scala.tools.nsc.typechecker.Namers$LockingTypeCompleter.complete(Namers.scala:2039)
	at scala.tools.nsc.typechecker.Namers$LockingTypeCompleter.complete$(Namers.scala:2037)
	at scala.tools.nsc.typechecker.Namers$PolyTypeCompleter.complete(Namers.scala:2051)
	at scala.tools.nsc.typechecker.Namers$CompleterWrapper.complete(Namers.scala:2088)
	at scala.tools.nsc.typechecker.Namers$Namer$$anon$1.complete(Namers.scala:616)
	at scala.reflect.internal.Symbols$Symbol.info(Symbols.scala:1542)
	at scala.reflect.internal.Symbols$Symbol.initialize(Symbols.scala:1690)
	at scala.tools.nsc.typechecker.Typers$Typer.$anonfun$typedStats$6(Typers.scala:3325)
	at scala.tools.nsc.typechecker.Typers$Typer.$anonfun$typedStats$6$adapted(Typers.scala:3323)
	at scala.Option$WithFilter.foreach(Option.scala:274)
	at scala.tools.nsc.typechecker.Typers$Typer.$anonfun$typedStats$4(Typers.scala:3323)
	at scala.tools.nsc.typechecker.Typers$Typer.$anonfun$typedStats$4$adapted(Typers.scala:3321)
	at scala.reflect.internal.Scopes$Scope.foreach(Scopes.scala:415)
	at scala.tools.nsc.typechecker.Typers$Typer.addSynthetics$1(Typers.scala:3321)
	at scala.tools.nsc.typechecker.Typers$Typer.typedStats(Typers.scala:3383)
	at scala.tools.nsc.typechecker.Typers$Typer.typedTemplate(Typers.scala:2030)
	at scala.tools.nsc.typechecker.Typers$Typer.typedClassDef(Typers.scala:1843)
	at scala.tools.nsc.typechecker.Typers$Typer.typed1(Typers.scala:5646)
	at scala.tools.nsc.typechecker.Typers$Typer.typed(Typers.scala:5726)
	at scala.tools.nsc.typechecker.Typers$Typer.typedStat$1(Typers.scala:5790)
	at scala.tools.nsc.typechecker.Typers$Typer.$anonfun$typedStats$10(Typers.scala:3373)
	at scala.tools.nsc.typechecker.Typers$Typer.typedStats(Typers.scala:3373)
	at scala.tools.nsc.typechecker.Typers$Typer.typedPackageDef$1(Typers.scala:5356)
	at scala.tools.nsc.typechecker.Typers$Typer.typed1(Typers.scala:5649)
	at scala.tools.nsc.typechecker.Typers$Typer.typed(Typers.scala:5726)
	at scala.tools.nsc.typechecker.Typers$Typer.typedStat$1(Typers.scala:5790)
	at scala.tools.nsc.typechecker.Typers$Typer.$anonfun$typedStats$10(Typers.scala:3373)
	at scala.tools.nsc.typechecker.Typers$Typer.typedStats(Typers.scala:3373)
	at scala.tools.nsc.typechecker.Typers$Typer.typedPackageDef$1(Typers.scala:5356)
	at scala.tools.nsc.typechecker.Typers$Typer.typed1(Typers.scala:5649)
	at scala.tools.nsc.typechecker.Typers$Typer.typed(Typers.scala:5726)
	at scala.tools.nsc.typechecker.Analyzer$typerFactory$TyperPhase.apply(Analyzer.scala:114)
	at scala.tools.nsc.Global$GlobalPhase.applyPhase(Global.scala:448)
	at scala.tools.nsc.interactive.Global$TyperRun.$anonfun$applyPhase$1(Global.scala:1340)
	at scala.tools.nsc.interactive.Global$TyperRun.applyPhase(Global.scala:1340)
	at scala.tools.nsc.interactive.Global$TyperRun.typeCheck(Global.scala:1333)
	at scala.tools.nsc.interactive.Global.typeCheck(Global.scala:665)
	at scala.tools.nsc.interactive.Global.$anonfun$backgroundCompile$7(Global.scala:580)
	at scala.tools.nsc.interactive.Global.$anonfun$backgroundCompile$6(Global.scala:576)
	at scala.tools.nsc.interactive.Global.$anonfun$backgroundCompile$6$adapted(Global.scala:576)
	at scala.collection.TraversableLike$WithFilter.$anonfun$foreach$1(TraversableLike.scala:792)
	at scala.collection.immutable.List.foreach(List.scala:392)
	at scala.collection.TraversableLike$WithFilter.foreach(TraversableLike.scala:791)
	at scala.tools.nsc.interactive.Global.backgroundCompile(Global.scala:576)
	at scala.tools.nsc.interactive.PresentationCompilerThread.run(PresentationCompilerThread.scala:32)

code navigation does not work for the file '/Users/gabro/buildo/config-scala/.metals/readonly/cats/data/OptionT.scala' because it doesn't belong to a build target.
```

</p>
</details>

--- a/metals/src/main/scala/scala/meta/internal/metals/Compilers.scala
+++ b/metals/src/main/scala/scala/meta/internal/metals/Compilers.scala
@@ -104,33 +104,46 @@ class Compilers(
       params: CompletionParams,
       token: CancelToken
   ): Option[CompletionList] =
-    withPC(params) { (pc, pos) =>
+    withPC(params, None) { (pc, pos) =>
       pc.complete(
         CompilerOffsetParams(pos.input.syntax, pos.input.text, pos.start, token)
       )
     }
   def hover(
       params: TextDocumentPositionParams,
-      token: CancelToken
+      token: CancelToken,
+      interactiveSemanticdbs: InteractiveSemanticdbs
   ): Option[Optional[Hover]] =
-    withPC(params) { (pc, pos) =>
+    withPC(params, Some(interactiveSemanticdbs)) { (pc, pos) =>
       pc.hover(
         CompilerOffsetParams(pos.input.syntax, pos.input.text, pos.start, token)
       )
     }
   def signatureHelp(
       params: TextDocumentPositionParams,
-      token: CancelToken
+      token: CancelToken,
+      interactiveSemanticdbs: InteractiveSemanticdbs
   ): Option[SignatureHelp] =
-    withPC(params) { (pc, pos) =>
+    withPC(params, Some(interactiveSemanticdbs)) { (pc, pos) =>
       pc.signatureHelp(
         CompilerOffsetParams(pos.input.syntax, pos.input.text, pos.start, token)
       )
     }
 
-  private def loadCompiler(path: AbsolutePath): Option[PresentationCompiler] = {
+  def loadCompiler(
+      path: AbsolutePath,
+      interactiveSemanticdbs: Option[InteractiveSemanticdbs]
+  ): Option[PresentationCompiler] =
     for {
-      target <- buildTargets.inverseSources(path)
+      target <- buildTargets
+        .inverseSources(path)
+        .orElse(interactiveSemanticdbs.flatMap(_.getBuildTarget(path)))
+      compiler <- loadCompiler(target)
+    } yield compiler
+  def loadCompiler(
+      target: BuildTargetIdentifier
+  ): Option[PresentationCompiler] = {
+    for {
       info <- buildTargets.info(target)
       scala <- info.asScalaBuildTarget
       isSupported = ScalaVersions.isSupportedScalaVersion(
@@ -157,10 +170,11 @@ class Compilers(
   }
 
   private def withPC[T](
-      params: TextDocumentPositionParams
+      params: TextDocumentPositionParams,
+      interactiveSemanticdbs: Option[InteractiveSemanticdbs]
   )(fn: (PresentationCompiler, Position) => T): Option[T] = {
     val path = params.getTextDocument.getUri.toAbsolutePath
-    loadCompiler(path).map { compiler =>
+    loadCompiler(path, interactiveSemanticdbs).map { compiler =>
       val input = path.toInputFromBuffers(buffers)
       val pos = params.getPosition.toMeta(input)
       val result = fn(compiler, pos)

--- a/metals/src/main/scala/scala/meta/internal/metals/InteractiveSemanticdbs.scala
+++ b/metals/src/main/scala/scala/meta/internal/metals/InteractiveSemanticdbs.scala
@@ -1,24 +1,21 @@
 package scala.meta.internal.metals
 
 import ch.epfl.scala.bsp4j.BuildTargetIdentifier
-import ch.epfl.scala.bsp4j.ScalacOptionsItem
 import java.net.URI
 import java.nio.charset.Charset
-import java.nio.file.Files
 import java.nio.file.Paths
 import java.util.Collections
-import java.util.concurrent.TimeUnit
 import java.util.concurrent.atomic.AtomicReference
 import org.eclipse.lsp4j.DiagnosticSeverity
 import org.eclipse.lsp4j.PublishDiagnosticsParams
 import org.eclipse.{lsp4j => l}
 import scala.collection.concurrent.TrieMap
 import scala.concurrent.ExecutionContext
-import scala.meta.interactive.InteractiveSemanticdb
 import scala.meta.internal.io.FileIO
 import scala.meta.internal.metals.MetalsEnrichments._
 import scala.meta.internal.mtags.Semanticdbs
 import scala.meta.internal.mtags.TextDocumentLookup
+import scala.meta.internal.semanticdb.TextDocument
 import scala.meta.internal.tokenizers.PlatformTokenizerCache
 import scala.meta.internal.{semanticdb => s}
 import scala.meta.io.AbsolutePath
@@ -40,7 +37,8 @@ final class InteractiveSemanticdbs(
     client: MetalsLanguageClient,
     tables: Tables,
     messages: Messages,
-    statusBar: StatusBar
+    statusBar: StatusBar,
+    compilers: () => Compilers
 )(implicit ec: ExecutionContext)
     extends Cancelable
     with Semanticdbs {
@@ -135,39 +133,26 @@ final class InteractiveSemanticdbs(
     }
   }
 
-  private def compile(source: AbsolutePath): Option[s.TextDocument] = {
-    for {
-      buildTarget <- getBuildTarget(source)
-      global <- Option(
-        globalCache.computeIfAbsent(buildTarget, x => newGlobal(x).orNull)
-      )
-    } yield {
-      val text = FileIO.slurp(source, charset)
-      val uri = source.toURI.toString
-      val textDocument = InteractiveSemanticdb
-        .toTextDocument(
-          global,
-          code = text,
-          filename = uri,
-          timeout = TimeUnit.SECONDS.toMillis(15),
-          options = List(
-            "-P:semanticdb:synthetics:on",
-            "-P:semanticdb:symbols:none",
-            "-P:semanticdb:text:on"
-          )
-        )
-        .withUri(uri) // semanticdb-scalac does weird URI encoding
-      textDocumentCache.put(source, textDocument)
-      PlatformTokenizerCache.megaCache.clear() // :facepalm:
-      textDocument
-    }
-  }
-
-  private def getBuildTarget(
+  def getBuildTarget(
       source: AbsolutePath
   ): Option[BuildTargetIdentifier] = {
     val fromDatabase = tables.dependencySources.getBuildTarget(source)
     fromDatabase.orElse(inferBuildTarget(source))
+  }
+
+  private def compile(source: AbsolutePath): Option[s.TextDocument] = {
+    for {
+      buildTarget <- getBuildTarget(source)
+      pc <- compilers().loadCompiler(buildTarget)
+    } yield {
+      val text = FileIO.slurp(source, charset)
+      val uri = source.toURI.toString
+      val bytes = pc.semanticdbTextDocument(uri, text)
+      val textDocument = TextDocument.parseFrom(bytes)
+      textDocumentCache.put(source, textDocument)
+      PlatformTokenizerCache.megaCache.clear() // :facepalm:
+      textDocument
+    }
   }
 
   private def inferBuildTarget(
@@ -184,45 +169,5 @@ final class InteractiveSemanticdbs(
       id
     }
   }.take(1).toList.headOption
-
-  private def newGlobal(buildTarget: BuildTargetIdentifier): Option[Global] = {
-    for {
-      info <- buildTargets.info(buildTarget)
-      scalaInfo <- info.asScalaBuildTarget
-      if {
-        val isOk = scalaInfo.getScalaVersion.startsWith("2.12")
-        if (!isOk) reportUnsupportedScalaVersion(scalaInfo.getScalaVersion)
-        isOk
-      }
-      scalacOptions <- buildTargets.scalacOptions(buildTarget)
-    } yield newGlobal(scalacOptions)
-  }
-
-  private def reportUnsupportedScalaVersion(scalaVersion: String): Unit = {
-    statusBar.addMessage(Only212Navigation.statusBar(scalaVersion))
-    val notification = tables.dismissedNotifications.Only212Navigation
-    if (!notification.isDismissed) {
-      notification.dismiss(2, TimeUnit.MINUTES)
-      client
-        .showMessageRequest(Only212Navigation.params(scalaVersion))
-        .asScala
-        .foreach { item =>
-          if (item == Only212Navigation.dismissForever) {
-            notification.dismissForever()
-          }
-        }
-    }
-  }
-
-  private def newGlobal(item: ScalacOptionsItem): Global = {
-    val classpath = item.getClasspath.asScala.iterator
-      .map(uri => Paths.get(URI.create(uri)))
-      .filterNot(path => Files.isDirectory(path))
-      .mkString(java.io.File.pathSeparator)
-    val scalacOptions = item.getOptions.asScala.iterator
-      .filterNot(_.isNonJVMPlatformOption)
-      .toList
-    InteractiveSemanticdb.newCompiler(classpath, scalacOptions)
-  }
 
 }

--- a/metals/src/main/scala/scala/meta/internal/metals/Messages.scala
+++ b/metals/src/main/scala/scala/meta/internal/metals/Messages.scala
@@ -136,32 +136,6 @@ class Messages(icons: Icons) {
     }
   }
 
-  object Only212Navigation {
-    def statusBar(scalaVersion: String) =
-      MetalsStatusParams(
-        "$(alert) No navigation",
-        tooltip = params(scalaVersion).getMessage
-      )
-    def dismissForever: MessageActionItem =
-      new MessageActionItem("Don't show again")
-    def ok: MessageActionItem =
-      new MessageActionItem("Ok")
-    def params(scalaVersion: String): ShowMessageRequestParams = {
-      val params = new ShowMessageRequestParams()
-      params.setMessage(
-        s"Navigation for external library sources is not supported in Scala $scalaVersion."
-      )
-      params.setType(MessageType.Warning)
-      params.setActions(
-        List(
-          ok,
-          dismissForever
-        ).asJava
-      )
-      params
-    }
-  }
-
   object SelectBspServer {
     case class Request(
         params: ShowMessageRequestParams,

--- a/metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala
+++ b/metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala
@@ -175,7 +175,8 @@ class MetalsLanguageServer(
         languageClient,
         tables,
         messages,
-        statusBar
+        statusBar,
+        () => compilers
       )
     )
     warnings = new Warnings(
@@ -683,7 +684,7 @@ class MetalsLanguageServer(
   @JsonRequest("textDocument/hover")
   def hover(params: TextDocumentPositionParams): CompletableFuture[Hover] =
     CancelTokens { token =>
-      compilers.hover(params, token) match {
+      compilers.hover(params, token, interactiveSemanticdbs) match {
         case None => null
         case Some(value) =>
           value.orElse(null)
@@ -829,7 +830,7 @@ class MetalsLanguageServer(
       params: TextDocumentPositionParams
   ): CompletableFuture[SignatureHelp] =
     CancelTokens { token =>
-      compilers.signatureHelp(params, token).orNull
+      compilers.signatureHelp(params, token, interactiveSemanticdbs).orNull
     }
 
   @JsonRequest("textDocument/codeAction")

--- a/mtags-interfaces/src/main/java/scala/meta/pc/PresentationCompiler.java
+++ b/mtags-interfaces/src/main/java/scala/meta/pc/PresentationCompiler.java
@@ -55,6 +55,11 @@ public abstract class PresentationCompiler {
      */
     public abstract Optional<Hover> hover(OffsetParams params);
 
+    /**
+     * Returns the Protobuf byte array representation of a SemanticDB <code>TextDocument</code> for the given source.
+     */
+    public abstract byte[] semanticdbTextDocument(String filename, String code);
+
     // =================================
     // Configuration and lifecycle APIs.
     // =================================

--- a/mtags/src/main/scala/scala/meta/internal/pc/ScalaPresentationCompiler.scala
+++ b/mtags/src/main/scala/scala/meta/internal/pc/ScalaPresentationCompiler.scala
@@ -13,6 +13,7 @@ import org.eclipse.lsp4j.Hover
 import org.eclipse.lsp4j.SignatureHelp
 import scala.collection.JavaConverters._
 import scala.concurrent.ExecutionContext
+import scala.meta.internal.metals.EmptyCancelToken
 import scala.meta.pc.OffsetParams
 import scala.meta.pc.PresentationCompiler
 import scala.meta.pc.SymbolSearch
@@ -112,6 +113,20 @@ case class ScalaPresentationCompiler(
     ) { global =>
       Optional.ofNullable(new HoverProvider(global, params).hover().orNull)
     }
+
+  override def semanticdbTextDocument(
+      filename: String,
+      code: String
+  ): Array[Byte] = {
+    access.withNonCancelableCompiler(
+      Array.emptyByteArray,
+      EmptyCancelToken
+    ) { global =>
+      new SemanticdbTextDocumentProvider(global)
+        .textDocument(filename, code)
+        .toByteArray
+    }
+  }
 
   def newCompiler(): MetalsGlobal = {
     val classpath = this.classpath.mkString(File.pathSeparator)

--- a/mtags/src/main/scala/scala/meta/internal/pc/SemanticdbTextDocumentProvider.scala
+++ b/mtags/src/main/scala/scala/meta/internal/pc/SemanticdbTextDocumentProvider.scala
@@ -1,0 +1,32 @@
+package scala.meta.internal.pc
+
+import scala.meta.internal.semanticdb.scalac.SemanticdbConfig
+import scala.meta.internal.{semanticdb => s}
+
+class SemanticdbTextDocumentProvider(val compiler: MetalsGlobal) {
+  import compiler._
+  def textDocument(
+      filename: String,
+      code: String
+  ): s.TextDocument = {
+    val unit = addCompilationUnit(
+      code = code,
+      filename = filename,
+      cursor = None
+    )
+    typeCheck(unit)
+    semanticdbOps.config = SemanticdbConfig.parse(
+      List(
+        "-P:semanticdb:synthetics:on",
+        "-P:semanticdb:symbols:none",
+        "-P:semanticdb:text:on"
+      ),
+      _ => (),
+      compiler.reporter,
+      SemanticdbConfig.default
+    )
+    import semanticdbOps._
+    val document = unit.toTextDocument
+    document.withUri(filename)
+  }
+}

--- a/tests/slow/src/test/scala/tests/DefinitionCrossSlowSuite.scala
+++ b/tests/slow/src/test/scala/tests/DefinitionCrossSlowSuite.scala
@@ -1,0 +1,33 @@
+package tests
+
+object DefinitionCrossSlowSuite
+    extends BaseCompletionSlowSuite("definition-cross") {
+  testAsync("2.11") {
+    cleanDatabase()
+    for {
+      _ <- server.initialize(
+        """
+          |/metals.json
+          |{
+          |  "a": {
+          |    "scalaVersion": "2.11.12"
+          |  }
+          |}
+          |/a/src/main/scala/a/Main.scala
+          |object Main {
+          |  println("hello!")
+          |}
+          |""".stripMargin
+      )
+      _ = client.messageRequests.clear()
+      _ <- server.didOpen("a/src/main/scala/a/Main.scala")
+      _ = server.workspaceDefinitions // trigger definition
+      _ <- server.didOpen("scala/Predef.scala")
+      _ = assertNoDiff(
+        client.workspaceMessageRequests,
+        ""
+      )
+      _ = assertNoDiff(client.workspaceDiagnostics, "")
+    } yield ()
+  }
+}

--- a/tests/unit/src/main/scala/tests/TestingClient.scala
+++ b/tests/unit/src/main/scala/tests/TestingClient.scala
@@ -166,8 +166,6 @@ final class TestingClient(workspace: AbsolutePath, buffers: Buffers)
           ImportBuildChanges.yes
         } else if (params == ImportBuild.params) {
           ImportBuild.yes
-        } else if (params == Only212Navigation.params("2.11.12")) {
-          Only212Navigation.dismissForever
         } else if (CheckDoctor.isDoctor(params)) {
           CheckDoctor.moreInformation
         } else if (SelectBspServer.isSelectBspServer(params)) {

--- a/tests/unit/src/main/scala/tests/TestingServer.scala
+++ b/tests/unit/src/main/scala/tests/TestingServer.scala
@@ -371,13 +371,14 @@ final class TestingServer(
 
   private def offsetParams(
       filename: String,
-      original: String
+      original: String,
+      root: AbsolutePath
   ): Future[(String, TextDocumentPositionParams)] = {
     val offset = original.indexOf("@@")
     if (offset < 0) sys.error(s"missing @@\n$original")
     val text = original.replaceAllLiterally("@@", "")
     val input = m.Input.String(text)
-    val path = workspace.resolve(filename)
+    val path = root.resolve(filename)
     path.touch()
     val pos = m.Position.Range(input, offset, offset)
     for {
@@ -396,10 +397,11 @@ final class TestingServer(
   def assertHover(
       filename: String,
       query: String,
-      expected: String
+      expected: String,
+      root: AbsolutePath = workspace
   ): Future[Unit] = {
     for {
-      hover <- hover(filename, query)
+      hover <- hover(filename, query, root)
     } yield {
       DiffAssertions.assertNoDiffOrPrintObtained(
         hover,
@@ -410,9 +412,13 @@ final class TestingServer(
     }
   }
 
-  def hover(filename: String, query: String): Future[String] = {
+  def hover(
+      filename: String,
+      query: String,
+      root: AbsolutePath
+  ): Future[String] = {
     for {
-      (text, params) <- offsetParams(filename, query)
+      (text, params) <- offsetParams(filename, query, root)
       hover <- server.hover(params).asScala
     } yield TestHovers.renderAsString(text, Option(hover), includeRange = false)
   }

--- a/tests/unit/src/test/scala/tests/DefinitionSlowSuite.scala
+++ b/tests/unit/src/test/scala/tests/DefinitionSlowSuite.scala
@@ -2,7 +2,6 @@ package tests
 
 import scala.concurrent.Future
 import scala.concurrent.duration.Duration
-import scala.meta.internal.metals.Messages.Only212Navigation
 import scala.meta.internal.metals.MetalsServerConfig
 import scala.meta.internal.metals.StatisticsConfig
 
@@ -231,35 +230,6 @@ object DefinitionSlowSuite extends BaseSlowSuite("definition") {
       )
       _ <- server.didFocus("a/src/main/scala/a/Main.scala")
       // dependency diagnosticsForDebuggingPurposes are unpublished.
-      _ = assertNoDiff(client.workspaceDiagnostics, "")
-    } yield ()
-  }
-
-  testAsync("2.11") {
-    cleanDatabase()
-    for {
-      _ <- server.initialize(
-        """
-          |/metals.json
-          |{
-          |  "a": {
-          |    "scalaVersion": "2.11.12"
-          |  }
-          |}
-          |/a/src/main/scala/a/Main.scala
-          |object Main {
-          |  println("hello!")
-          |}
-          |""".stripMargin
-      )
-      _ = client.messageRequests.clear()
-      _ <- server.didOpen("a/src/main/scala/a/Main.scala")
-      _ = server.workspaceDefinitions // trigger definition
-      _ <- server.didOpen("scala/Predef.scala")
-      _ = assertNoDiff(
-        client.workspaceMessageRequests,
-        Only212Navigation.params("2.11.12").getMessage
-      )
       _ = assertNoDiff(client.workspaceDiagnostics, "")
     } yield ()
   }

--- a/tests/unit/src/test/scala/tests/HoverSlowSuite.scala
+++ b/tests/unit/src/test/scala/tests/HoverSlowSuite.scala
@@ -1,5 +1,7 @@
 package tests
 
+import scala.meta.internal.metals.Directories
+
 object HoverSlowSuite extends BaseSlowSuite("hover") with TestHovers {
 
   testAsync("basic") {
@@ -16,6 +18,32 @@ object HoverSlowSuite extends BaseSlowSuite("hover") with TestHovers {
           |  Option(1).he@@ad
           |}""".stripMargin,
         """override def head: Int""".hover
+      )
+    } yield ()
+  }
+
+  testAsync("dependencies") {
+    for {
+      _ <- server.initialize(
+        """/metals.json
+          |{"a":{}}
+          |/a/src/main/scala/a/Main.scala
+          |package a
+          |object Main {
+          |  println(42)
+          |}
+        """.stripMargin
+      )
+      _ <- server.didOpen("a/src/main/scala/a/Main.scala")
+      _ = server.workspaceDefinitions // triggers goto definition, creating Predef.scala
+      _ <- server.assertHover(
+        "scala/Predef.scala",
+        """
+          |object Main {
+          |  Option(1).he@@ad
+          |}""".stripMargin,
+        """override def head: Int""".hover,
+        root = workspace.resolve(Directories.readonly)
       )
     } yield ()
   }

